File: src/CSharp/1-100/73.SetMatrixZeroes.cs
using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.MatrixSetZeroes
{
public class Solution {
    public void SetZeroes(int[][] matrix) {
        bool isCol = false;
        int R = matrix.Length;
        int C = matrix[0].Length;

        for (int i = 0; i < R; i++) {
            if (matrix[i][0] == 0) {
                isCol = true;
            }

            for (int j = 1; j < C; j++) {
                if (matrix[i][j] == 0) {
                    matrix[0][j] = 0;
                    matrix[i][0] = 0;
                }
            }
        }
        for (int i = 1; i < R; i++) {
            for (int j = 1; j < C; j++) {
                if (matrix[i][0] == 0 || matrix[0][j] == 0) {
                    matrix[i][j] = 0;
                }
            }
        }
        if (matrix[0][0] == 0) {
            for (int j = 0; j < C; j++) {
                matrix[0][j] = 0;
            }
        }
        if (isCol) {
            for (int i = 0; i < R; i++) {
                matrix[i][0] = 0;
            }
        }
    }
}
}

File: src/Python/1-100/73.SetMatrixZeroes.py

class Solution:
    def setZeroes(self, matrix) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        m = len(matrix)
        n = len(matrix[0])
        rowLineSet = set()
        columnLineSet = set()
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 0:
                    rowLineSet.add(i)
                    columnLineSet.add(j)
        for i in range(m):
            for j in range(n):
                if i in rowLineSet or j in columnLineSet:
                    matrix[i][j] = 0
        return matrix

if __name__ == "__main__":
    solution = Solution()
    print(solution.setZeroes([
        [1, 1, 1],
        [1, 0, 1],
        [1, 1, 1]
    ]))

   